### PR TITLE
[CF-181, CF-202] Fix a problem with duplicate images

### PR DESCRIPTION
**Pull request by SVilgelm**
Filter the deleted images from destination cloud.
Create an image with new id in case the image with same id exists.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.image.test_glance_image
tests/cloudferrylib/os/image/test_glance_image.py:50: [W0212(protected-access), GlanceImageTestCase.setUp] Access to a protected member _resp of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/image/glance_image.py [hunk 1]
```diff
@@ -196,14 +196,23 @@ def get_image_list(self):
...
         return images
 
     def create_image(self, **kwargs):
-        return self.glance_client.images.create(**kwargs)
+        try:
+            return self.glance_client.images.create(**kwargs)
+        except glance_exceptions.HTTPConflict:
+            image_id = kwargs.pop('id', None)
+            if image_id is not None:
+                LOG.debug("Retry to create an image '%s' (%s) without old id",
```
Comment (roman-verchikov) [inline cloudferrylib/os/image/glance_image.py]: Let's have this message as warning, and change message to something like: "Image ID will not be kept for source image %(id)s, image ID is already present in destination (perhaps was deleted previously)."

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.image.test_glance_image
tests/cloudferrylib/os/image/test_glance_image.py:50: [W0212(protected-access), GlanceImageTestCase.setUp] Access to a protected member _resp of a client class

File: cloudferrylib/os/image/glance_image.py (hunk 1 above)
Comment (SVilgelm) [inline cloudferrylib/os/image/glance_image.py]: Done

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.image.test_glance_image
tests/cloudferrylib/os/image/test_glance_image.py:50: [W0212(protected-access), GlanceImageTestCase.setUp] Access to a protected member _resp of a client class

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1, can be merged

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.image.test_glance_image
tests/cloudferrylib/os/image/test_glance_image.py:50: [W0212(protected-access), GlanceImageTestCase.setUp] Access to a protected member _resp of a client class

**Pull request merged**